### ISSUE TITLE
Model always return [1]

### ISSUE THREAD

**Issue by ElieTaillard** [cadswetha/Intelligent-farming-system#1]
Hello, 
Your code is really cool but I have a question about the model. 
I have tried several values but unfortunately the model always returns `[1]`. 
No matter what I put on, I can't get the model to return to me `[0]`.

Because the prediction always returns `[1]`, the `else` statement in this piece of code is never executed https://github.com/cadswetha/Intelligent-farming-system/blob/945475cbd57d89cbee4ae93e09e89e3a7e7bfdbf/app.py#L52-L57

**Comment by cadswetha** [comment]
> Hello, Your code is really cool but I have a question about the model. I have tried several values but unfortunately the model always returns `[1]`. No matter what I put on, I can't get the model to return to me `[0]`.
> 
> Because the prediction always returns `[1]`, the `else` statement in this piece of code is never executed
> 
> https://github.com/cadswetha/Intelligent-farming-system/blob/945475cbd57d89cbee4ae93e09e89e3a7e7bfdbf/app.py#L52-L57

**Comment by cadswetha** [comment]
Thank you for bringing this to my attention. As the developer responsible for the front-end part, I will work with my teammate to address the issue and ensure that it is resolved quickly.

**Comment by ElieTaillard** [comment]
I found the issue.
I just deleted these lines:
https://github.com/cadswetha/Intelligent-farming-system/blob/945475cbd57d89cbee4ae93e09e89e3a7e7bfdbf/create_model.py#L24-L25
I'll try to look further but from my first tests it seems to fix the issue.

**Comment by cadswetha** [comment]
Oh so we don't have to normalize our data then?
Not sure why it works now
I'll look into it as well

Also, checking for the moisture value in the first if condition is actually not needed.

**Comment by ElieTaillard** [comment]
Yes I agree this if should be removed. The AI should predict itself if the irragation is needed or not
https://github.com/cadswetha/Intelligent-farming-system/blob/945475cbd57d89cbee4ae93e09e89e3a7e7bfdbf/app.py#L52-L53

**Comment by ElieTaillard** [comment]
I wrote a python script that tests the model.
Here is the code:
```
import pickle
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

# Load model
with open('RandomForest.pkl', 'rb') as f:
    model = pickle.load(f)

test_file_path = 'datasets.csv'

# Load data
df_test = pd.read_csv(test_file_path)

# Data prep
X_test = df_test.drop('Irrigation', axis=1)
y_test = df_test['Irrigation']

# Predict
y_pred = model.predict(X_test)

# Create dataframe to store predictions
predictions_df = pd.DataFrame({'Prediction': y_pred})

# Show predictions
print(predictions_df)

# Add predictions in test Dataframe
df_test_with_predictions = df_test.assign(Prediction=y_pred)

# Show dataframe with the predictions
print(df_test_with_predictions)

# Save dataframe as .csv
df_test_with_predictions.to_csv('predictions.csv', index=False)

# Show difference between expected and predicted
data = pd.read_csv('predictions.csv')

col1 = data['Irrigation']
col2 = data['Prediction']

difference = col2 - col1

plt.plot(difference)
plt.xlabel('Index')
plt.ylabel('Difference')
plt.title('Difference between expected and predicted\n (values that are not equal to 0 represent bad predictions))')
plt.show()
```
Here is the result with the lines:
```
 Xtrain = scaler.fit_transform(Xtrain) 
 Xtest = scaler.transform(Xtest) 
```
![image](https://user-images.githubusercontent.com/54487782/220944078-dec4a2d2-20ca-4b7a-80d7-869147b6a42e.png)

You can see that predictions are really bad.

Now without the two lines:
![image](https://user-images.githubusercontent.com/54487782/220944447-5a57f572-b333-4285-8599-02408558d443.png)

You can see that predictions are much better.

**Comment by cadswetha** [comment]
Woah!! There's a huge difference

How did you find out that those those lines are the actual cause of the problem?

**Comment by ElieTaillard** [comment]
I asked ChatGPT to generate a script to train a model using the RandomForestClassifier model.
He listed the steps and in the code he provided I did not see the two lines you have in your script. I tried to remove these lines to see if they were useful. And I saw that it fixed the issue. 😅

**Comment by cadswetha** [comment]
Ahhh got it!! Thanks to you and ChatGPT 😅
Now I'll have to check why normalizing the data didn't work.

**Comment by ElieTaillard** [comment]
From what I understand, the model is trained with normalized data.
Maybe you have to run the `predict()` function with normalized data, I don't know.

https://github.com/cadswetha/Intelligent-farming-system/blob/945475cbd57d89cbee4ae93e09e89e3a7e7bfdbf/create_model.py#L42-L45
https://github.com/cadswetha/Intelligent-farming-system/blob/945475cbd57d89cbee4ae93e09e89e3a7e7bfdbf/app.py#L50-L51
